File: src/dataset/cpp/raindrops_Exercism_cmccandless/claude/ClaudeSonnet4_raindrops_v4.cpp
#include "raindrops.h"

string raindrops::convert(int x)
{
   string result;
   
   const bool div3 = x % 3 == 0;
   const bool div5 = x % 5 == 0;
   const bool div7 = x % 7 == 0;
   
   if (div3 | div5 | div7) {
      result.reserve(15);
      if (div3) result += "Pling";
      if (div5) result += "Plang";
      if (div7) result += "Plong";
   } else {
      result = std::to_string(x);
   }
   
   return result;
}

File: src/dataset/go/raindrops_Exercism_go_thinkverse/claude/ClaudeSonnet4_raindrops_v4.go
package raindrops

import (
	"strconv"
)

func Convert(number int) string {
	div3 := number%3 == 0
	div5 := number%5 == 0
	div7 := number%7 == 0

	if !(div3 || div5 || div7) {
		return strconv.Itoa(number)
	}

	var result [15]byte
	pos := 0

	if div3 {
		copy(result[pos:], "Pling")
		pos += 5
	}

	if div5 {
		copy(result[pos:], "Plang")
		pos += 5
	}

	if div7 {
		copy(result[pos:], "Plong")
		pos += 5
	}

	return string(result[:pos])
}
